extmod/network_cyw43.c

```diff
@@ -121,8 +121,16 @@ STATIC mp_obj_t network_cyw43_active(size_t n_args, const mp_obj_t *args) {
     if (n_args == 1) {
         return mp_obj_new_bool(cyw43_tcpip_link_status(self->cyw, self->itf));
     } else {
+        #ifdef hal_check_network_allowed
+        if (mp_obj_is_true(args[1])) {
+            hal_check_network_allowed(MP_QSTR_WLAN);
+        }
+        #endif
         uint32_t country = CYW43_COUNTRY(mod_network_country_code[0], mod_network_country_code[1], 0);
         cyw43_wifi_set_up(self->cyw, self->itf, mp_obj_is_true(args[1]), country);
+        #ifdef hal_set_network_active
+        hal_set_network_active(MP_QSTR_WLAN, mp_obj_is_true(args[1]));
+        #endif
```

Comment:
This needs to take into account both STA and AP.  Either or both can be enabled, but you only want to release access to the network (ie allow BT) if both STA and AP are turned off.

Reply:
Yes - I've changed to use STA/AP and I'm using a bit mask so STA and AP are handled properly.